services/cronjob/dataproxyrefresher/cmd/dataproxyrefresher/main.go

```diff
@@ -25,231 +26,421 @@ func init() {
 	global.Init()
 }
 
-func cleanUp() {
-	mongodb.Client.Disconnect()
-	os.Exit(0)
+const (
+	// SchemaName represents the name of the schema being used.
+	SchemaName = "karte_von_morgen-v1.0.0"
+	// APIEntry is the base endpoint for the API from which profiles are
+	// retrieved.
+	APIEntry = "https://api.ofdb.io/v0/entries/"
+
+	// APIValidatePath is the API path used for validating profiles.
+	APIValidatePath = "/v2/validate"
+	// APINodesPath is the API path used for operations related to nodes.
+	APINodesPath = "/v2/nodes"
+	// APIProfilesPath is the API path used for operations related to profiles.
+	APIProfilesPath = "/v1/profiles"
+)
+
+type DataproxyRefresher struct {
+	svc service.ProfilesService
 }
 
 func main() {
-	schemaName := "karte_von_morgen-v1.0.0"
-	apiEntry := "https://api.ofdb.io/v0/entries/"
+	logger.Info("Start DataproxyRefresher...")
+	startTime := time.Now()
 
-	svc := service.NewProfileService(
-		mongo.NewProfileRepository(mongodb.Client.GetClient()),
-	)
+	refresher := NewDataproxyRefresher()
+
+	if err := refresher.Run(); err != nil {
+		logger.Error("Error running DataproxyRefresher", err)
+		os.Exit(1)
+		return
+	}
+
+	duration := time.Since(startTime)
+	logger.Info("DataproxyRefresher has finished")
+	logger.Info("DataproxyRefresher run duration: " + duration.String())
+}
+
+func NewDataproxyRefresher() *DataproxyRefresher {
+	return &DataproxyRefresher{
+		svc: service.NewProfileService(
+			mongo.NewProfileRepository(mongodb.Client.GetClient()),
+		),
+	}
+}
+
+func (r *DataproxyRefresher) Run() error {
+	defer r.cleanUp()
+
+	profiles, err := r.getProfiles(SchemaName)
+	if err != nil {
+		return err
+	}
+
+	mapping, err := r.getMapping(SchemaName)
+	if err != nil {
+		return err
+	}
 
+	for _, profile := range profiles {
+		if err := r.processProfile(mapping, profile); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *DataproxyRefresher) getProfiles(
+	schemaName string,
+) ([]model.Profile, error) {
 	curTime := time.Now().Unix()
 	refreshBefore := curTime - config.Conf.RefreshTTL
-	profiles, err := svc.FindLessThan(schemaName, refreshBefore)
+
+	profiles, err := r.svc.FindLessThan(schemaName, refreshBefore)
 	if err != nil {
 		if err == mongodb.ErrNoDocuments {
-			logger.Error("No profile found.", err)
-		} else {
-			logger.Error("Failed to find data from profiles.", err)
+			return nil, fmt.Errorf("no profile found: %w", err)
 		}
-		cleanUp()
+		return nil, fmt.Errorf("failed to find data from profiles: %w", err)
 	}
 
-	// get mapping
+	return profiles, nil
+}
+
+func (r *DataproxyRefresher) getMapping(
+	schemaName string,
+) (map[string]string, error) {
 	mapping, err := importutil.GetMapping(schemaName)
 	if err != nil {
-		logger.Error("Failed to get mapping.", err)
-		cleanUp()
+		return nil, fmt.Errorf("failed to get mapping: %w", err)
```

Comment:
@TingSyuanWang, the new code will exit with a status of 1 if there's any error, including failure to find the mapping. I'm not sure if we should change the behavior, such as ignoring the not found errors.